misc_scripts/add_allele_registry_link.rb

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env ruby
+
+require "uri"
+require 'digest/sha1'
+
+
+# create URL with authentication
+# params = vector of param=value
+def createUrl2(hostPath, params, login, password)
```

Comment:
For both createUrl methods, you could potentially use URI's builder method and ActiveSupport's to_query method to handle all of this automatically. See

https://stackoverflow.com/a/33257393
and
https://ruby-doc.org/stdlib-1.9.3/libdoc/uri/rdoc/URI/HTTP.html#method-c-build-label-Description

`to_query` should handle the escaping as needed for you, but will only be available if you move this to `lib` so `ActiveSupport` is in the library path.

Up to you though - if this works, it works and rewriting it might not be worth it.